src/common/providers/identity.ts

```diff
@@ -1044,23 +818,43 @@ function wrapHandler(
         logger.error('Invalid request, unable to process');
         throw new HttpsError('invalid-argument', 'Bad Request');
       }
-      const rawDecodedJWT = decodeJWT(req.body.data.jwt);
-      const decodedPayload = shouldVerifyJWT()
-        ? verifyJWT(req.body.data.jwt, rawDecodedJWT, keysCache)
-        : (rawDecodedJWT.payload as DecodedPayload);
-      checkDecodedToken(decodedPayload, eventType, projectId);
+
+      if (!apps().admin.auth()._verifyAuthBlockingToken) {
+        throw new Error(
+          'Cannot validate Auth Blocking token. Please update Firebase Admin SDK to >= v10.1.0'
+        );
+      }
+
+      const decodedPayload: DecodedPayload = isDebugFeatureEnabled(
+        'skipTokenVerification'
+      )
+        ? unsafeDecodeAuthBlockingToken(req.body.data.jwt)
+        : await apps()
+            .admin.auth()
+            ._verifyAuthBlockingToken(req.body.data.jwt);
+
       const authUserRecord = parseAuthUserRecord(decodedPayload.user_record);
       const authEventContext = parseAuthEventContext(decodedPayload, projectId);
-      const authResponse =
-        (await handler(authUserRecord, authEventContext)) || undefined;
+
+      const authResponse = v2
```

Comment:
nit* I wonder if you can do this by checking the length of arguments expected by the handler rather than using additional argument e.g.

https://github.com/firebase/firebase-functions/blob/master/src/common/providers/https.ts#L736-L746

(This is arguably less clear what's going on)

Reply:
Yeah that makes sense, i'll change it so we follow the same pattern